I compared the clang compile warnings on MacOS for the devel and suggest_override_final branch.  I was hoping there would be fewer warnings but it looks like some new ones cropped up (below). Probably not significant since these are all compiler 'suggestions' but nevertheless thought worth noting.

```
warning: unknown warning option '-Wsuggest-attribute=format'; did you mean '-Wproperty-attribute-mismatch'? [-Wunknown-warning-option]
warning: unknown warning option '-Wsuggest-attribute=noreturn'; did you mean '-Wbounds-attributes-redundant'? [-Wunknown-warning-option]
warning: unknown warning option '-Wsuggest-final-methods'; did you mean '-Wsuggest-override'? [-Wunknown-warning-option]
warning: unknown warning option '-Wsuggest-final-types' [-Wunknown-warning-option]
warning: unknown warning option '-Wvolatile' [-Wunknown-warning-option]
In file included from /Users/kgriesser/work/rev/src/RevExt.cc:11:
In file included from /Users/kgriesser/work/rev/include/RevExt.h:15:
/Users/kgriesser/work/rev/include/SST.h:21:32: warning: unknown warning group '-Wsuggest-final-methods', ignored [-Wunknown-warning-option]
   21 | #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
      |                                ^
/Users/kgriesser/work/rev/include/SST.h:22:32: warning: unknown warning group '-Wsuggest-final-types', ignored [-Wunknown-warning-option]
   22 | #pragma GCC diagnostic ignored "-Wsuggest-final-types"
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ else()
   set(FP_MODE_FLAG "-frounding-math -ftrapping-math")
 endif()
 
-set(CMAKE_CXX_FLAGS "-std=c++17 ${FP_MODE_FLAG} -O2 -Wall -Wextra ${WERROR_FLAG} -Wsuggest-override -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wvolatile -Wvla -Wuninitialized -Wdouble-promotion -Wno-unused-parameter -Wno-Wmismatched-tags -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+set(CMAKE_CXX_FLAGS "-std=c++17 ${FP_MODE_FLAG} -O2 -Wall -Wextra ${WERROR_FLAG} -Wsuggest-override -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wvolatile -Wvla -Wuninitialized -Wdouble-promotion -Wno-unused-parameter -Wno-mismatched-tags -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -Wall ${REVCPU_COMPILER_MACROS}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall ${REVCPU_COMPILER_MACROS}")
 

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -83,7 +83,7 @@ else()
   set(FP_MODE_FLAG "-frounding-math -ftrapping-math")
 endif()
 
-set(CMAKE_CXX_FLAGS "-std=c++17 ${FP_MODE_FLAG} -O2 -Wall -Wextra ${WERROR_FLAG} -Wsuggest-override -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wvolatile -Wvla -Wuninitialized -Wdouble-promotion -Wno-unused-parameter -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
+set(CMAKE_CXX_FLAGS "-std=c++17 ${FP_MODE_FLAG} -O2 -Wall -Wextra ${WERROR_FLAG} -Wsuggest-override -Wsuggest-attribute=format -Wsuggest-attribute=noreturn -Wsuggest-final-methods -Wsuggest-final-types -Wvolatile -Wvla -Wuninitialized -Wdouble-promotion -Wno-unused-parameter -Wno-Wmismatched-tags -Wno-deprecated-declarations ${CMAKE_CXX_FLAGS} -I./ ${LDFLAGS} ${REVCPU_COMPILER_MACROS}")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -O0 -Wall ${REVCPU_COMPILER_MACROS}")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3 -Wall ${REVCPU_COMPILER_MACROS}")
 

--- a/include/RevCSR.h
+++ b/include/RevCSR.h
@@ -61,7 +61,14 @@ namespace SST::RevCPU {
 class RevCore;
 
 struct RevCSR : RevZicntr {
-  static constexpr size_t CSR_LIMIT = 0x1000;
+  static constexpr size_t CSR_LIMIT  = 0x1000;
+
+  RevCSR()                           = default;
+  RevCSR( const RevCSR& )            = delete;
+  RevCSR( RevCSR&& )                 = default;
+  RevCSR& operator=( const RevCSR& ) = delete;
+  RevCSR& operator=( RevCSR&& )      = delete;
+  virtual ~RevCSR()                  = default;
 
   // CSR Registers
   enum : uint16_t {
@@ -549,10 +556,7 @@ private:
   std::array<uint64_t, CSR_LIMIT>                                         CSR{};     ///< RegCSR: CSR registers
   std::unordered_map<uint16_t, std::function<uint64_t( uint16_t )>>       Getter{};  ///< RevCSR: CSR Getters
   std::unordered_map<uint16_t, std::function<bool( uint16_t, uint64_t )>> Setter{};  ///< RevCSR: CSR Setters
-
-  /// RevCSR: Get the core owning this hart
-  virtual RevCore* GetCore() const = 0;
-};  // class RevCSR
+};                                                                                   // class RevCSR
 
 }  // namespace SST::RevCPU
 

--- a/include/RevRegFile.h
+++ b/include/RevRegFile.h
@@ -192,7 +192,7 @@ public:
   RevRegFile& operator=( const RevRegFile& ) = delete;
 
   /// RevRegFile: standard destructor
-  ~RevRegFile()                              = default;
+  ~RevRegFile() final                        = default;
 
   ///< RevRegFile: Return the core owning this hart
   RevCore* GetCore() const final { return Core; }

--- a/include/RevZicntr.h
+++ b/include/RevZicntr.h
@@ -23,6 +23,7 @@ class RevCore;
 class RevZicntr {
   uint64_t InstRet{};  ///< RevZicntr: Number of instructions retired
 
+protected:
   /// RevZicntr: Get the core owning this hart
   virtual RevCore* GetCore() const = 0;
 
@@ -40,7 +41,12 @@ class RevZicntr {
     return make_dependent<T>( GetCore() )->output->fatal( CALL_INFO, -1, msg, GetPC() );
   }
 
-protected:
+  RevZicntr()                              = default;
+  RevZicntr( const RevZicntr& )            = delete;
+  RevZicntr( RevZicntr&& )                 = default;
+  RevZicntr& operator=( const RevZicntr& ) = delete;
+  RevZicntr& operator=( RevZicntr&& )      = delete;
+
   template<typename ZICNTR>
   static uint64_t rdcycle( const ZICNTR* Zicntr ) {
     return Zicntr->GetCore()->GetCycles();

--- a/include/SST.h
+++ b/include/SST.h
@@ -18,6 +18,7 @@
 #pragma GCC diagnostic ignored "-Wnon-virtual-dtor"
 #pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #pragma GCC diagnostic ignored "-Wsuggest-override"
+#pragma GCC diagnostic ignored "-Wdouble-promotion"
 #pragma GCC diagnostic ignored "-Wsuggest-final-methods"
 #pragma GCC diagnostic ignored "-Wsuggest-final-types"
 

--- a/scripts/slurm/build-gcc11-sst13.1.0.sh
+++ b/scripts/slurm/build-gcc11-sst13.1.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc11-sst14.0.0.sh
+++ b/scripts/slurm/build-gcc11-sst14.0.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc11-sst14.1.0.sh
+++ b/scripts/slurm/build-gcc11-sst14.1.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc11.sh
+++ b/scripts/slurm/build-gcc11.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc13-sst13.1.0.sh
+++ b/scripts/slurm/build-gcc13-sst13.1.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc13-sst14.0.0.sh
+++ b/scripts/slurm/build-gcc13-sst14.0.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-gcc13.sh
+++ b/scripts/slurm/build-gcc13.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-llvm12-sst13.1.0.sh
+++ b/scripts/slurm/build-llvm12-sst13.1.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-llvm12-sst14.0.0.sh
+++ b/scripts/slurm/build-llvm12-sst14.0.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-llvm12-sst14.1.0.sh
+++ b/scripts/slurm/build-llvm12-sst14.1.0.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

--- a/scripts/slurm/build-llvm12.sh
+++ b/scripts/slurm/build-llvm12.sh
@@ -35,7 +35,7 @@ cd build || exit
 rm -Rf ./*
 
 #-- Stage 3: initiate the build
-cmake -DBUILD_ASM_TESTING=ON -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
+cmake -DCMAKE_BUILD_TYPE=Debug -DRVCC=${RVCC} ../
 make clean
 make uninstall
 make -j

